vochain/proof.go

```diff
@@ -89,8 +89,12 @@ func VerifyProofOffChainTree(process *models.Process, proof *models.Proof,
 		if err != nil {
 			return false, nil, fmt.Errorf("cannot hash proof key: %w", err)
 		}
-		valid, err := tree.VerifyProof(hashFunc, hashedKey, []byte{}, p.Siblings, censusRoot)
-		return valid, bigOne, err
+		valid, err := tree.VerifyProof(hashFunc, hashedKey, p.Value, p.Siblings, censusRoot)
+		// Legacy: support p.Value == nil, assume then value=1
+		if p.Value == nil {
+			return valid, bigOne, err
+		}
+		return valid, new(big.Int).SetBytes(p.Value), err
```

Comment:
This is converting bytes to bigint interpreting bytes as big-endian, when those bytes come from inside the merkletree where those same bytes are treated as little-endian.
Maybe would make sense to update this line to return the bigint interepreting the bytes as little-endian, as are some bytes coming from inside the tree (which works in little-endian).

In fact, arbo already has a method that abstracts from 'endianness' to convert from bytes to bigint and viceversa, using that, this line could be:
```suggestion
		return valid, arbo.BytesToBigInt(p.Value), err
```

Reply:
So, there is something I don't understand.

In order to add the value, we do `err := tr.Add(data, r.CensusValue)`. The r.CensusValue is a slice of bytes. Are you saying that Arbo will switch the order of that slice internally on tr.Add() but when returning the value it will just return the bytes (without reversing them again)?

If that's the case I don't understand the logic behind it, thb.

Reply:
The thing is that if arbo is using a blake2b or sha256 hash internally, the endianness is not a problem. But, if the instance of arbo used is with Poseidon hash, arbo inputs are still `[]byte`, but later internally will need to be converted into `*big.Int` in order to compute the Poseidon hashes. So arbo interprets those byte arrays as little-endian in order to convert them to bigints (only in the Poseidon hash case): https://github.com/vocdoni/arbo/blob/master/tree.go#L278

Now, if we use the `Add` method without taking in account the endianness, we could input a byte-array that encodes a bigint in big-endian, and when internally is converted to bigint to compute the Poseidon hash it gives an error due the value being outside of the finite field.
As an example, the value `21888242871839275222246405745257275088548364400416034343698204186575808495616` (which is inside the finite field).
If we encode it as big-endian it is `0x30644e72e131a029b85045b68181585d2833e84879b9709143e1f593effff449` (in big-endian hex). If we give those bytes, internally when being interpreted as little-endian will give the value `33451715034712113185045486448192360878076553814222909500690766407339250443312`, which was not the original one, but that's not the only problem: when trying to compute the Poseidon hash of this value, it will give the error `inputs values not inside Finite Field`.

As an iteration on the previous example, if somebody sends a vote with `weight` equal to `11888242871839275222246405745257275088548364400416034343698204186575808499697` encoded as a big-endian byte array, when the Vochain tries to add it into a Poseidon Tree, it will not be able and it will return the mentioned error.